test/integration/mecanum_drive_system.cc

```diff
@@ -0,0 +1,537 @@
+            pub.Publish(msg);
+          });
+
+      server.AddSystem(velocityRamp.systemPtr);
+
+      server.Run(true, 3000, false);
+
+      // Poses for 4s
+      ASSERT_EQ(4000u, poses.size());
+
+      int sleep = 0;
+      int maxSleep = 30;
+      for (; odomPoses.size() < 150 && sleep < maxSleep; ++sleep)
+      {
+        std::this_thread::sleep_for(std::chrono::milliseconds(100));
+      }
+      ASSERT_NE(maxSleep, sleep);
+
+      // Odometry calculates the pose of a point that is located half way
+      // between the two wheels, not the origin of the model. For example,
+      // if the vehicle is commanded to rotate in place, the vehicle will
+      // rotate about the point half way between the two wheels, thus,
+      // the odometry position will remain zero.
+      // However, since the model origin is offset, the model position will
+      // change. To find the final pose of the model, we have to do the
+      // following similarity transformation
+      
+      math::Pose3d tOdomModel(0.554283,0,-0.325,0,0,0);
+      auto finalModelFramePose =
+          tOdomModel * odomPoses.back() * tOdomModel.Inverse();
+
+      // Odom for 3s
+      ASSERT_FALSE(odomPoses.empty());
+      EXPECT_EQ(150u, odomPoses.size());
+
+      auto expectedLowerPosition =
+          (forward ? poses[0].Pos() : poses[3999].Pos());
+      auto expectedGreaterPosition =
+          (forward ? poses[3999].Pos() : poses[0].Pos());
+
+      EXPECT_LT(expectedLowerPosition.X(), expectedGreaterPosition.X());
+      EXPECT_LT(expectedLowerPosition.Y(), expectedGreaterPosition.Y());
+      EXPECT_NEAR(expectedLowerPosition.Z(), expectedGreaterPosition.Z(), tol);
+      EXPECT_NEAR(poses[0].Rot().X(), poses[3999].Rot().X(), tol);
+      EXPECT_NEAR(poses[0].Rot().Y(), poses[3999].Rot().Y(), tol);
+      EXPECT_NEAR(poses[0].Rot().Z(), poses[3999].Rot().Z(), tol);
+
+      // The value from odometry will be close, but not exactly the ground truth
+      // pose of the robot model. This is partially due to throttling the
+      // odometry publisher, partially due to a frame difference between the
+      // odom frame and the model frame, and partially due to wheel slip.
+      EXPECT_NEAR(poses.back().Pos().X(), odomPoses.back().Pos().X(), 1e-2);
+      EXPECT_NEAR(poses.back().Pos().Y(), odomPoses.back().Pos().Y(), 1e-2);
+      // EXPECT_NEAR(poses.back().Pos().X(), finalModelFramePose.Pos().X(), 1e-2);
+      // EXPECT_NEAR(poses.back().Pos().Y(), finalModelFramePose.Pos().Y(), 1e-2);
+
+      // Verify velocity and acceleration boundaries.
+      // Moving time.
+      double t = 3.0;
+      double d = poses[3999].Pos().Distance(poses[0].Pos());
+      double v0 = 0;
+      double v = d / t;
+      double a = (v - v0) / t;
+
+      EXPECT_LT(v, 0.5);
+      EXPECT_LT(a, 1);
+      EXPECT_GT(v, -0.5);
+      EXPECT_GT(a, -1);
+
+    };
+
+  testCmdVel(true /*test forward movement*/);
+  testCmdVel(false /*test backward movement*/);
```

Comment:
I was unaware that this MR existed and basically did the exact same work two weeks ago. I also refactored the `TestPublishCmd` function to also test the sideways, diagonal and rotational movement of the drive instead of only forward and backward, which is what the diff drive test does.

Feel free to grab the function code from my changes [in here](https://github.com/glpuga/gz-sim/commit/f676237824a5c94b8c51c9c293728cc83b6cf7cd#diff-849a4ca12fab1a6ce3db85103424fabf832712f141e89a883f9289f81c285761R224-R235)  to improve the coverage.